File: php/features.php
<?php
declare(strict_types=1);

// FriedApps SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FriedAppsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FriedAppsBaseFeature();
            case "test":
                return new FriedAppsTestFeature();
            default:
                return new FriedAppsBaseFeature();
        }
    }
}

File: py/friedapps_sdk/features.py
# FriedApps SDK feature factory

from friedapps_sdk.feature.base_feature import FriedAppsBaseFeature
from friedapps_sdk.feature.test_feature import FriedAppsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FriedAppsBaseFeature(),
        "test": lambda: FriedAppsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
